content.en/tools/nebula/_index.md

```diff
@@ -0,0 +1,90 @@
+
+Nebula calculates when the next probe is due with the following formula
+
+```
+NOW + floor(max_interval, ceil(min_interval, 1.2 * (NOW - previous_successful_dial))
+```
+
+The maximum interval is set to 15m and the minimum interval is set to 1m.
+
+The monitoring process doesn't establish a proper libp2p connection (which involves the protocol negotiation) but only dials the peer on a transport level by calling [`DialPeer`](https://github.com/libp2p/go-libp2p/blob/8d771355b41297623e05b04a865d029a2522a074/p2p/net/swarm/swarm_dial.go#L229). It also liberally retries dialing peers if errors occur [[source](https://github.com/dennis-tra/nebula/blob/a20481d35509411b03b3fcfcf0f148b49a5eda3f/pkg/monitor/dialer.go#L117)].
+
+## What data does Nebula gather?
+
+The crawler component establishes a proper libp2p connection to the remote peer. This means Nebula and the peer exchange the list of supported protocols and user agent information. Further, to connect to the peer Nebula must have known the network addresses of the remote peer. It also measures the latency to dial, connect, and crawl. The dial latency just includes the establishment of a connection on the transport level and the connect latency also includes the protocol handshake. The crawl latency measures how long it took to extract information from other peers _k_-buckets.
+
+To summarize, Nebula gathers the following information about all peers it was able to connect to:
+
+- peer ID
+
+- user agent
+
+- supported protocols
+
+- all advertised Multiaddresses
+
+- connection latencies (dial, connect, crawl - see above)
+
+- potential connection errors
+
+The monitoring component periodically checks if peers are still online. This allows us to additionally measure sessions of uptime for peers.
+
+Because we crawl and probe the network periodically, and because Multiaddresses contain IP addresses we can also answer the following questions:
+
+- When do peers update their IPFS node?
+
+- How long has a peer been online?
+
+- In which country/city is a peer located? (powered by [Maxmind](https://www.maxmind.com/en/home))
+
+- Does the peer run in a data center?(powered by [Udger](https://udger.com/))
+
+On top of the above, Nebula also tracks _neighbor_ information. We consider peers in _k_-buckets to be neighbors of the peer who maintains these _k_-buckets. This information spans a graph where each node is a peer and each edge corresponds to a _k_-bucket entry.
+
+## Deployment
+
+We are running the crawler to measure the following networks:
+
+| Network  | Crawl Frequency  | Crawl duration |
+| -------- | ---------------- | -------------- |
+| IPFS     | every 30 minutes | ~5 minutes     |
+| Filecoin | every 30 minutes | ~1 minute      |
+| Polkadot | every 30 minutes | ~25 minutes    |
```

Comment:
I think it's around 10k peers. The reason it takes so long is because, only 1-2k are actually reachable. The connection attempts to the rest are just timing out after 30s (or a minute?). This adds up because we only do so many connection attempts in parallel.